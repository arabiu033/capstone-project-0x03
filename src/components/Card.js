import PropTypes from 'prop-types';
import React from 'react';
import lit from '../assets/images/lit.png';
import Pea from '../assets/images/Pea.png';
import phy from '../assets/images/phy.png';
import med from '../assets/images/med.png';
import eco from '../assets/images/eco.png';
import che from '../assets/images/che.png';

export const images = {
  lit,
  Pea,
  phy,
  med,
  eco,
  che,
};

const Card = (props) => {
  const { laureates } = props;
  const img = laureates.meta.nobelPrizeCategory;
  return (
    laureates.laureates.map((lau) => (
      <article key={lau.id} className="card">
        <div className="category-icon">
          <img src={images[img] || images.Pea} alt={img} />
        </div>
        <section className="category-date">
          <p>Date Awarded</p>
          <span className="divider" />
          <p>{lau.nobelPrizes[0].dateAwarded || lau.nobelPrizes[0].awardYear}</p>
        </section>
        <h3 className="category-name">{(lau.fullName && lau.fullName.en) || (lau.orgName.en)}</h3>
        <p>
          {'$ '}
          {lau.nobelPrizes[0].prizeAmount}
        </p>
        <section className="more-info">
          <p className="motivation">{lau.nobelPrizes[0].motivation.en}</p>
          <a href={lau.wikipedia.english} rel="noreferrer" target="_blank" className="wiki"><h5>More (wiki)</h5></a>
        </section>
      </article>
    ))
  );
};

Card.propTypes = {
  laureates: PropTypes.arrayOf(PropTypes.elementType).isRequired,
};

export default Card;
