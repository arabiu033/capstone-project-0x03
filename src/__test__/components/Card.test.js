import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';

import Card from '../../components/Card';

test('renders correctly Header component', () => {
  const card = renderer.create(
    <MemoryRouter>
      <Card />
    </MemoryRouter>,

  ).toJSON();
  expect(card).toMatchSnapshot();
});
