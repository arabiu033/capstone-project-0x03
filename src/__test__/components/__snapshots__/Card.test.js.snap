// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly Header component 1`] = `null`;
